src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Eggress/AzureBlob/AzureBlobEgressProviderTests.cs

```diff
@@ -0,0 +1,295 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Azure.Storage.Sas;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.TestCommon.Fixtures;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Tools.Monitor.Egress;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.AzureBlob;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests.Eggress.AzureBlob
+{
+    [TargetFrameworkMonikerTrait(TargetFrameworkMonikerExtensions.CurrentTargetFrameworkMoniker)]
+    public class AzureBlobEgressProviderTests : IClassFixture<AzuriteFixture>, IDisposable
+    {
+        private readonly ITestOutputHelper _outputHelper;
+        private readonly AzuriteFixture _azuriteFixture;
+        private readonly TemporaryDirectory _tempDirectory;
+
+        private readonly string _testUploadFile;
+        private readonly string _testUploadFileHash;
+
+        public AzureBlobEgressProviderTests(ITestOutputHelper outputHelper, AzuriteFixture azuriteFixture)
+        {
+            _outputHelper = outputHelper;
+            _azuriteFixture = azuriteFixture;
+            _tempDirectory = new TemporaryDirectory(outputHelper);
+
+            _testUploadFile = Path.Combine(_tempDirectory.FullName, Path.GetRandomFileName());
+            File.WriteAllText(_testUploadFile, "Sample Contents\n123");
+            _testUploadFileHash = GetFileSHA256(_testUploadFile);
+        }
+
+        [ConditionalFact(Timeout = TestTimeouts.EgressUnitTestTimeoutMs)]
+        public async Task AzureBlobEgress_UploadsCorrectData()
+        {
+            _azuriteFixture.SkipTestIfNotAvailable();
+
+            // Arrange
+            TestOutputLoggerProvider loggerProvider = new(_outputHelper);
+            AzureBlobEgressProvider egressProvider = new(loggerProvider.CreateLogger<AzureBlobEgressProvider>());
+
+            BlobContainerClient containerClient = await ConstructBlobContainerClientAsync(create: false);
+
+            AzureBlobEgressProviderOptions providerOptions = ConstructEgressProviderSettings(containerClient);
+            EgressArtifactSettings artifactSettings = ConstructArtifactSettings();
+
+            // Act
+            string identifier = await egressProvider.EgressAsync(providerOptions, UploadAction, artifactSettings, CancellationToken.None);
+
+            // Assert
+            List<BlobItem> blobs = await GetAllBlobsAsync(containerClient);
+            Assert.Single(blobs);
```

Comment:
nit: `Assert.Single` returns the strongly typed instance of the single object if it passes the assertion. You can combine the next line with this line so that you don't have to call the `First` extension method.